### ISSUE TITLE
a way to /useblock using ~ ~ ~ ,sntead of fix coordinates*

### ISSUE THREAD

**Issue by EgeSisli** [MCCTeam/Minecraft-Console-Client#1815]
so i havea farm using minecarts i constantly afk with 3 afk accounts with minecarts constantly right clicking the block in front of them so i cant just sat /useblock x y z and repeat i need to use smt like /useblock ~ ~1 ~1 to make it work bc the block theyre interacting constantly changes kinda like this but when i tried using ~ 1~ ~1 mcc crashed

**Comment by xXjojaXx** [comment]
U could use a chatbot and get the location of the bot via `Location location = GetCurrentLocation();` and retrieve the X, Y and Z Coordinates like this: `double x = location.X;`. If you're looking to use the coordinates as part of an Internal Command like in your question, you could do it the following way: `PerformInternalCommand($"useblock {x} {y} {z + 1}");`. Please correct me if I made an mistake, the code isn't tested. Also if you have any problems or questions , feel free to ask them.

PS: I think your example crashed, because mcc doesn't have the Minecraft-Logic of relative coordinates included, so it's very likely that it tried to parse/convert `~1` to an number, which, of course isn't possible

**Comment by TyraVex** [comment]
This option would be awesome. +1

**Comment by TyraVex** [comment]
Unfortunately, @xXjojaXx 's solution doesn't work...

```
[MCC] [Script] System.InvalidOperationException: Cannot access protected member `MinecraftClient.ChatBot.GetCurrentLocation()' via a qualifier of type `MinecraftClient.ChatBot'. The qualifier must be of type `ScriptLoader.Script' or derived from it
```

Is there any workarounds for this error?

**Comment by TyraVex** [comment]
> Unfortunately, @xXjojaXx 's solution doesn't work (for me at least)...
> 
> `` [MCC] [Script] System.InvalidOperationException: Cannot access protected member `MinecraftClient.ChatBot.GetCurrentLocation()' via a qualifier of type `MinecraftClient.ChatBot'. The qualifier must be of type `ScriptLoader.Script' or derived from it ``
> 
> Is there any workarounds for this error?

**Comment by TyraVex** [comment]
Nvm it's working i'm a noob lol

//MCCScript 1.0

MCC.LoadBot(new Test());

//MCCScript Extensions

class Test : ChatBot
{
  public override void Initialize()
  {
   Location location = GetCurrentLocation();
   double x = location.X;
   double y = Math.Round(location.Y - 1);
   double z = location.Z;
   LogToConsole(x + " " + y + " " + z);
   if (x < 0) {
     x = Math.Round(x - 0.5);
   } else {
     x = Math.Round(z);
   }
   if (z < 0) {
     z = Math.Round(z - 0.5);
   } else {
     z = Math.Round(z);
   }
   PerformInternalCommand("useblock " + x + " " + y + " " + z);
  }
}

**Comment by milutinke** [comment]
@BruceChenQAQ You could re-use the logic you added for this for `/move`

**Comment by BruceChenQAQ** [comment]
Implemented in https://github.com/MCCTeam/Minecraft-Console-Client/commit/85f6d807bf3933e2724afb8f7f3581795fa30aa7.

**Comment by milutinke** [comment]
> Implemented in [85f6d80](https://github.com/MCCTeam/Minecraft-Console-Client/commit/85f6d807bf3933e2724afb8f7f3581795fa30aa7).

Awesome
